systemtest/src/test/java/io/strimzi/systemtest/oauth/OauthST.java

```diff
@@ -0,0 +1,203 @@
+ */
+package io.strimzi.systemtest.oauth;
+
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.openshift.client.OpenShiftClient;
+import io.strimzi.api.kafka.model.CertSecretSourceBuilder;
+import io.strimzi.systemtest.Constants;
+import io.strimzi.systemtest.MessagingBaseST;
+import io.strimzi.systemtest.annotations.OpenShiftOnly;
+import io.strimzi.systemtest.utils.StUtils;
+import io.strimzi.test.TestUtils;
+import io.strimzi.test.executor.Exec;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.codec.BodyCodec;
+import io.vertx.junit5.VertxExtension;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static io.strimzi.systemtest.Constants.REGRESSION;
+import static io.strimzi.systemtest.Resources.deployKeycloakNodePortService;
+
+@OpenShiftOnly
+@Tag(REGRESSION)
+@ExtendWith(VertxExtension.class)
+public class OauthST extends MessagingBaseST {
+
+    public static final String NAMESPACE = "oauth2-cluster-test";
+    protected static final Logger LOGGER = LogManager.getLogger(OauthST.class);
+
+    protected static final String TOPIC_NAME = "my-topic";
+    static final String REVERSE_TOPIC_NAME = "my-topic-reversed";
+
+    static final String PRODUCER_USER_NAME = "hello-world-producer";
+    static final String CONSUMER_USER_NAME = "hello-world-consumer";
+    static final String STREAMS_USER_NAME = "hello-world-streams";
+
+    protected WebClient client;
+    static String oauthTokenEndpointUri;
+    static String validIssuerUri;
+    static String jwksEndpointUri;
+    static String userNameClaim;
+
+    JsonObject sendHttpRequests(JsonObject records, String bridgeHost) throws InterruptedException, ExecutionException, TimeoutException {
+        LOGGER.info("Sending records to Kafka Bridge");
+        CompletableFuture<JsonObject> future = new CompletableFuture<>();
+        client.post(bridgeHost, "/topics/" + TOPIC_NAME)
+                .putHeader("Content-length", String.valueOf(records.toBuffer().length()))
+                .putHeader("Content-Type", Constants.KAFKA_BRIDGE_JSON_JSON)
+                .as(BodyCodec.jsonObject())
+                .sendJsonObject(records, ar -> {
+                    if (ar.succeeded()) {
+                        LOGGER.debug("Server accepted post");
+                        HttpResponse<JsonObject> response = ar.result();
+                        future.complete(response.body());
```

Comment:
it's not always in this way. Having `ar.succeeded` doesn't mean having HTTP 200 or 204. It just means that the request was ok but you need to check the HTTP status code anyway.